temporal-object/README.md

```diff
@@ -0,0 +1,167 @@
+  }
+
+  public boolean isExpired() {
+    // if expiration date before current date, return true
+    return getContract().getExpirationDate().compareTo(SimpleDate.getToday()) < 0;
+  }
+
+  public int getNumber() {
+    return getContract().getCardNumber();
+  }
+  //...
+}
+```
+
+Each `CreditContractVersion` consists of many variables, with the relevant parts
+
+```java 
+public class CreditContractVersion {
+  private final String contractText;
+  private final String contractIssuer;
+  private final String version;
+  private final int creditLimit;
+  private final int cardNumber;
+  private final int cvcCode;
+  private final SimpleDate expirationDate;
+}
+```
+
+Then, we can introduce an example application that makes use of these properties
+
+```java
+  public static void main(String[] args) {
+    SimpleDate.setToday(new SimpleDate(2000, 3, 1));
+
+    // start a credit card contract from today until 3rd of September 2002
+    CreditContractVersion firstContract = new CreditContractVersion("Original Contract",
+    "Bank of XYZ", "Version 1", 100, 12345, 1234,
+    new SimpleDate(2002, 9, 3));
+    CreditCard card = new CreditCard(firstContract, SimpleDate.getToday());
+    System.out.println("New card with limit of " + card.getCreditLimit()
+    + ", card number " + card.getNumber() + ", and cvc code of " + card.getCvc()
+    + " which expires on " + card.getExpiration());
+
+    // The card expires
+    SimpleDate.setToday(new SimpleDate(2002, 9, 18));
+    System.out.println(SimpleDate.getToday().toString() + " is expired: " + card.isExpired());
+
+    // And is then re-negotiated, with the new contract coming into effect a few days later
+
+    CreditContractVersion secondContract = new CreditContractVersion("new contract", "Bank of XYZ",
+    "Version 2", 120, 12345, 1234, new SimpleDate(2004, 3, 4));
+    card.addContract(secondContract, new SimpleDate(2002, 9, 21));
+
+    // A few days later when the new contract is effective
+    SimpleDate.setToday(new SimpleDate(2002, 9, 21));
+    System.out.println("Re-negotiated card with limit of " + card.getCreditLimit()
+    + ", card number " + card.getNumber() + ", and cvc code of " + card.getCvc()
+    + " which expires on " + card.getExpiration());
+    System.out.println(SimpleDate.getToday().toString() + " is expired: " + card.isExpired());
+    }
+}
+```
+
+Which has an output of 
+
+```
+New card with limit of 100, card number 12345, and cvc code of 1234 which expires on 2002, 9, 3
+2002, 9, 18 is expired: true
+Re-negotiated card with limit of 120, card number 12345, and cvc code of 1234 which expires on 2004, 3, 4
+2002, 9, 21 is expired: false
+```
+
+
+## Class diagram
+
+![alt text](./etc/temporal-object.urm.png "Audit Log")
+
+## Applicability
+
+Use the Temporal Object pattern when
+* Most of the objects properties are temporal.
+
+## Consequences
+
+A temporal object allows for individuals to refer explicit to versions, and temporal objects 
```

Comment:
Should this say "refer to explicit versions"?

Reply:
Thank you for catching this!